### PR TITLE
Update i18n recipe (Translate Routes, Add how to hide the default language in the URL) 

### PR DESCRIPTION
**Pull request by artursopelnik**
#### What kind of changes does this PR include?
- New or updated content

#### Description

Adds some important functions to i18n recipe.
- How to switch language between specific pages to get a better UX (Translate Routes).
- Instructions: If you don't want the default language to appear in the URL like the other languages.

#### CodeSandbox
https://sdvvp8-3000.csb.app/en/recipes/i18n/

#2894

### CODE REVIEW

**Comment by artursopelnik** [comment]
Mention @withastro/maintainers-docs

**Comment by ElianCodes** [comment]
I'm reviewing this one ASAP, but also pinging @Yan-Thomas as our i18n expert here

File: src/content/docs/en/recipes/i18n.mdx [hunk 1]
```diff
@@ -12,6 +12,8 @@ In this recipe, you'll learn how to use content collections and dynamic routing
 
 This example serves each language at its own subpath, e.g. `example.com/en/blog` for English and `example.com/fr/blog` for French.
 
+If you don't want the default language to show up in the URL like the other languages, you can use the [instructions](/en/recipes/i18n/#hide-default-language-in-the-url) below.
+
```
Comment (ElianCodes) [inline src/content/docs/en/recipes/i18n.mdx:16]:
```suggestion
If you prefer the default language to not be visible in the URL unlike other languages, there are [instructions to hide the default language](/en/recipes/i18n/#hide-default-language-in-the-url) below.

```

File: src/content/docs/en/recipes/i18n.mdx [hunk 2]
```diff
@@ -322,6 +324,179 @@ Create links to the different languages you support so users can choose the lang
     </html>
     ```
 
+### Hide default language in the URL
+
+1. Create a directory only for each language that is not the default language. For example, store your default language pages direclty in `pages/` and `fr/` if you also support French:
```
Comment (ElianCodes) [inline src/content/docs/en/recipes/i18n.mdx:329]:
```suggestion
1. Create a directory for each language except the default language. For example, store your default language pages directly in `pages/`, and your translated pages in `fr/`:
```

File: src/content/docs/en/recipes/i18n.mdx [hunk 3]
```diff
@@ -322,6 +324,179 @@ Create links to the different languages you support so users can choose the lang
...
+      - pages/
+        - about.astro
+        - index.astro
+        - **fr/**
+          - about.astro
+          - index.astro
+        - index.astro
+    </FileTree>
+
+2. Add one line to the `src/i18n/ui.ts` file to possibly toggle the feature:
```
Comment (ElianCodes) [inline src/content/docs/en/recipes/i18n.mdx:341]:
```suggestion
2. Add another line to the `src/i18n/ui.ts` file to toggle the feature:
```

File: src/content/docs/en/recipes/i18n.mdx [hunk 4]
```diff
@@ -322,6 +324,179 @@ Create links to the different languages you support so users can choose the lang
...
+
+    ```ts
+    // src/i18n/ui.ts
+    export const showDefaultLang = false;
+    ```
+
+
+3. Add helper function to the `src/i18n/utils.ts` file to translate paths based on the current language:
```
Comment (ElianCodes) [inline src/content/docs/en/recipes/i18n.mdx:349]:
```suggestion
3. Add a helper function to `src/i18n/utils.ts`, to translate paths based on the current language:
```

File: src/content/docs/en/recipes/i18n.mdx [hunk 5]
```diff
@@ -322,6 +324,179 @@ Create links to the different languages you support so users can choose the lang
...
+    import { ui, defaultLang, showDefaultLang } from './ui';
+
+    export function useTranslatedPath(lang: keyof typeof ui) {
+      return function translatePath(path: string, l: string = lang) {
+        return !showDefaultLang && l === defaultLang ? path : `/${l}${path}`
+      }
+    }
+    ```
+
+4. Import the helper where needed. For example, a nav component might look like:
```
Comment (ElianCodes) [inline src/content/docs/en/recipes/i18n.mdx:362]:
```suggestion
4. Import the helper where needed. For example, a `nav` component might look like:
```

File: src/content/docs/en/recipes/i18n.mdx [hunk 6]
```diff
@@ -322,6 +324,179 @@ Create links to the different languages you support so users can choose the lang
...
+        </li>
+        <li>
+            <a href="https://twitter.com/astrodotbuild">
+              {t('nav.twitter')}
+            </a>
+        </li>
+    </ul>
+    ```
+
+5. Helper could also be used to translate paths for specific language. For example, when users switch between languages:
```
Comment (ElianCodes) [inline src/content/docs/en/recipes/i18n.mdx:392]:
```suggestion
5. The helper function can also be used to translate paths for a specific language. For example, when users switch between languages:
```

File: src/content/docs/en/recipes/i18n.mdx [hunk 7]
```diff
@@ -322,6 +324,179 @@ Create links to the different languages you support so users can choose the lang
...
+      ))}
+    </ul>
+    ```
+
+### Translate Routes
+
+Translate the routes of your pages for each language.
+
+
+1. Add route mappings to the `src/i18n/ui.ts` file:
```
Comment (ElianCodes) [inline src/content/docs/en/recipes/i18n.mdx:414]:
```suggestion
Translate the routes of your pages for each language.

1. Add route mappings to `src/i18n/ui.ts`:
```

File: src/content/docs/en/recipes/i18n.mdx [hunk 8]
```diff
@@ -322,6 +324,179 @@ Create links to the different languages you support so users can choose the lang
...
+   de: {
+    'services': 'leistungen',
+   },
+   fr: {
+    'services': 'prestations-de-service',
+   },
+  }
+    ```
+
+2. Update the `useTranslatedPath` helper function in `src/i18n/utils.ts` to add some more route translation logic.
```
Comment (ElianCodes) [inline src/content/docs/en/recipes/i18n.mdx:428]:
```suggestion
2. Update the `useTranslatedPath` helper function in `src/i18n/utils.ts` to add router translation logic.
```

File: src/content/docs/en/recipes/i18n.mdx [hunk 9]
```diff
@@ -322,6 +324,179 @@ Create links to the different languages you support so users can choose the lang
...
+      const pathName = path.replaceAll('/', '')
+      const hasTranslation = defaultLang !== l && routes[l] !== undefined && routes[l][pathName] !== undefined
+      const translatedPath = hasTranslation ? '/' + routes[l][pathName] : path
+  
+      return !showDefaultLang && l === defaultLang ? translatedPath : `/${l}${translatedPath}`
+    }
+  }
+  ```
+
+3. Create a helper function to get the route if it exists based on the current URL in `src/i18n/utils.ts`:
```
Comment (ElianCodes) [inline src/content/docs/en/recipes/i18n.mdx:445]:
```suggestion
3. Create a helper function to get the route, if it exists based on the current URL, in `src/i18n/utils.ts`:
```

File: src/content/docs/en/recipes/i18n.mdx [hunk 10]
```diff
@@ -322,6 +324,179 @@ Create links to the different languages you support so users can choose the lang
...
+  
+    if (reversedKey !== undefined) {
+      return reversedKey
+    }
+  
+    return undefined
+  }
+  ```
+
+4. Helper could be used to get a translated route. For example, when users switch languages, they don't have to come back from the home page. However, if no route is defined, you will be taken to the home page
```
Comment (ElianCodes) [inline src/content/docs/en/recipes/i18n.mdx:481]:
```suggestion
4. The helper function can be used to get a translated route. For example, when no translated route is defined, the user will be redirected to the home page:
```

**Review by ElianCodes — COMMENTED** [review]
Added a couple suggestions for you to review

**Comment by artursopelnik** [comment]
@ElianCodes many thanks!

**Review by Yan-Thomas — APPROVED** [review]
This looks great, thanks @artursopelnik!

Note: @ElianCodes, when you merge this, please remove "en-only" from the stashed commit message confirmation if it happens to be there. This keyword is used for changes that are only relevant in English and should not mark other languages as outdated (e.g typo fixes)

**Pull request merged**